gptme/tools/base.py

```diff
@@ -33,9 +36,46 @@ class ExecuteFuncMsg(Protocol):
     def __call__(self, code: str, args: list[str], confirm: ConfirmFunc) -> Message: ...
 
 
+class ExecuteFn(Protocol):
+    def __call__(
+        self, args: dict[str, str], confirm: ConfirmFunc
+    ) -> Generator[Message, None, None] | Message: ...
+
+
```

Comment:
This is confusingly similarly named, we should find a way to reuse the `ExecuteFunc` protocol, probably by using it and mapping the parameters.

Reply:
I wasn't happy about the naming. Removed now.